File: XText46.cpp
#include "pt4.h"
#include <fstream>
using namespace std;

void Solve()
{
    Task("XText46");
    string s1, s2;
	pt >> s1 >> s2;
	ifstream f1(s1);
	ofstream f2(s2,ios::binary);
	
    while(f1.peek() != -1){
    	double num;
    	f1 >> num;
    	if(num!=(int)num && f1.peek() != -1){
			//f2 << (char *)&num;
			f2.write((char *)&num, sizeof(num));
			//f2.put(num);
			//put(num);
			//Show(num);
		}
	}
	f1.close();
	f2.close();
}

File: XText46.cs
// File: "XText46"
using PT4;
using System;
using System.IO;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace PT4Tasks
{
    public class MyTask : PT
    {
        public static void Solve()
        {
            Task("XText46");
            StreamReader f = new StreamReader(new FileStream(GetString(), FileMode.Open));
            BinaryWriter bw = new BinaryWriter(new FileStream(GetString(), FileMode.Create));
            while (!f.EndOfStream)
            {
                string line = f.ReadLine();
                var number = line.Split(' ');
                foreach (var num in number)
                {
                    var dou_num = double.TryParse(num, out var dou_var);
                    int int_var = (int)dou_var;
                    if (dou_num && int_var != dou_var)
                    {
                        bw.Write(dou_var);
                    }
                }
            }
            bw.Close();
        }

    }
}